TYPO3.Neos/Resources/Public/JavaScript/Content/Application.js

```diff
@@ -406,6 +406,12 @@ function(
 						$('title').html($htmlDom.filter('title').html());
 						$('link[rel="neos-site"]').attr('href', $htmlDom.filter('link[rel="neos-site"]').attr('href'));
 
+						var $pageStylesheets = $('link[rel="stylesheet"]:not(.neos-ui-asset)');
+						$pageStylesheets.remove();
+
+						var $loadedStylesheets = $htmlDom.filter('link[rel="stylesheet"]:not(.neos-ui-asset)');
+						$('head').append($loadedStylesheets);
```

Comment:
"more robust" is a joke in this context. Having a completely broken rendering is what this fixes, so I'd say it's always more robust. No?